lib/ecto/exceptions.ex

```diff
@@ -28,7 +28,7 @@ defmodule Ecto.QueryError do
   def exception(opts) do
     message = Keyword.fetch!(opts, :message)
     query   = Keyword.fetch!(opts, :query)
-    hint    = Keyword.fetch(opts, :hint)
+    hint    = Keyword.get(opts, :hint)
```

Comment:
I changed it here because I may pass `nil`, and if I do so, I wish it to be ignored. This won't happen with fetch + case, but will with get + if